internal/pkg/log/zap/zap.go

```diff
@@ -0,0 +1,126 @@
+// Package zap sets up a logger using the Zap logging framework.
+package zap
+
+import (
+	"io"
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Options contains all possible log settings.
+type Options struct {
+	// CallerSkip
+	CallerSkip int
+	// Level configures the log verbosity. Defaults to Debug.
+	Level *zap.AtomicLevel
+	// StacktraceLevel is the level which stacktraces will be emitted. Defaults
+	// to Warn.
+	StacktraceLevel *zap.AtomicLevel
+	// Encoder configures how Zap will encode the output. Defaults to JSON.
+	Encoder zapcore.Encoder
+	// OutputDest controls the destination of the log output. Defaults to
+	// os.Stderr.
+	OutputDest io.Writer
+	// ZapOptions allows passing additional optional zap.Options, ex: Sampling.
+	ZapOptions []zap.Option
+}
+
+// Opts allows manipulation of the Zap options.
+type Opts func(*Options)
```

Comment:
hmm, If you are going to create setters using this type, why does the `Options` struct need to have public attributes?

Reply:
It doesn't, good catch!